app/helpers/admin/terms_of_service_helper.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Admin
+  module TermsOfServiceHelper
+    def tos_need_accepting?
+      return false unless spree_user_signed_in?
+
+      return false if Spree::Config.enterprises_require_tos == false
+
+      !accepted_tos?
+    end
+
+    private
+
+    def accepted_tos?
+      file_uploaded_at = TermsOfServiceFile.updated_at
+
+      current_spree_user.terms_of_service_accepted_at.present? &&
+        current_spree_user.terms_of_service_accepted_at > file_uploaded_at &&
+        current_spree_user.terms_of_service_accepted_at < DateTime.now
```

Comment:
That's an interesting condition. So if I, for some weird reason, manage to tell the OFN database that I will accept future ToS until 3rd of March 2025 then this will be seen as invalid and it looks like I didn't accept the ToS. Until time passes and on the 3rd of March 2025, suddenly, I appear to have accepted the ToS. This is an impossible scenario at the moment but I wonder why you included it?

Reply:
Also, if I just accepted the ToS right now and our computers are so fast that the rendering happens in the same fraction of a second then it looks like I didn't accept the ToS?

Luckily timestamps are very accurate, although, the database is less accurate and therefore, within the accuracy of the database *now* would always be bigger than the stored timestamp. But I think that would need to be within a microsecond. Those zeros are nanoseconds:
```
Thu, 12 Oct 2023 10:51:46.765599000 AEDT +11:00
```

Reply:
David asked a similar question, see my answer here: https://github.com/openfoodfoundation/openfoodnetwork/pull/11795#discussion_r1395025415

I admit it's a bit overkilled.